core/src/main/java/org/apache/calcite/sql/validate/SqlUserDefinedAggFunction.java

```diff
@@ -54,7 +54,8 @@ public SqlUserDefinedAggFunction(SqlIdentifier opName,
       SqlOperandTypeInference operandTypeInference,
       SqlOperandTypeChecker operandTypeChecker, AggregateFunction function,
       boolean requiresOrder, boolean requiresOver, RelDataTypeFactory typeFactory) {
-    super(Util.last(opName.names), opName, SqlKind.OTHER_FUNCTION,
+    super(Util.last(opName.names), opName, SqlKind.getSqlKindByName(opName.names.get(0)) != null
+                    ? SqlKind.valueOf(opName.names.get(0)) : SqlKind.OTHER_FUNCTION,
```

Comment:
I'm not sure this is the right approach. What you do is you make "user-defined SUM function" to be treated as standard SUM function without giving a single check on the actual implementation.

That might trigger some optimization rules that alter results. For instance, Calcite could replace user-defined `AVG(...)` with standard `SUM(...)/COUNT(...)`. I'm not sure it is the right approach.